criu/kerndat.c

```diff
@@ -1056,63 +1056,63 @@ int kerndat_init(void)
 
 	ret = check_pagemap();
 	if (!ret)
-		ret = kerndat_get_shmemdev();
+		ret = kerndat_get_shmemdev(); BUG_ON(ret);
```

Comment:
Revert the BUG_ON messages. Add proper logging to the functions which need it. Don't use BUG_ON for expected runtime failures (it's meant for bugs in code).

Reply:
And please split those 2 patches.

Reply:
I'd suggested to keep both commits in one patch atleast for initial review with upstream because they're both in the same theme. And then if the maintainers request is, we'd split into separate PRs. 

Alternatively, we just send both PRs together just so the full picture is clear to them,

Reply:
With Angie we decided that they aren't actually that connected. One of them is about parsing command line arguments and the other is about initializing kerndat. Also since this PR is probably gonna grow and touch more files it's better to split it. Also one of them might be ready before the other and I think it's better to get patches merged as soon as possible instead of having one big patch in progress for a long time.